lib/autoload/duplicator.rb

```diff
@@ -1,15 +1,20 @@
 # frozen_string_literal: true
 class Duplicator
-  attr_reader :time_shift, :new_course_title
+  attr_reader :time_shift, :new_course_title, :current_user
+
   # Create an instance of Duplicator to track duplicated objects.
   #
   # @param [Enumerable] excluded_objects An Enumerable of objects to be excluded from duplication
   # @param [DateTime] time_shift The amount of time to shift dates by
-  def initialize(excluded_objects = [], time_shift = 0, new_course_title = 'Duplicated')
+  # @param [String] new_course_title The new course title
+  # @param [User] current_user Current user who initiated the duplication of the objects
+  def initialize(excluded_objects = [], time_shift = 0, new_course_title = 'Duplicated',
+                 current_user = User.system)
```

Comment:
We should default to `User.system` here since there is no context of a course in this model, but we can set the default of the `current_user` to be `course.creator` in the duplication service. 

Do you think that works?